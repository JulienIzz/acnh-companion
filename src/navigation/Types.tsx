export type RootStackParamList = {
  Home: undefined;
  Encyclopedie: undefined;
  Search: undefined;
  BugReporting: undefined;
};

export type Routes = keyof RootStackParamList;

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}
