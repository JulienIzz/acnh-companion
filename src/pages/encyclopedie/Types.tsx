export interface Fish {
  id: number;
  name: {
    'name-USen': string;
    'name-EUfr': string;
  };
  availability: {
    'month-northern': string;
    'month-southern': string;
    time: string;
    isAllDay: boolean;
    isAllYear: boolean;
    location: string;
    rarity: string;
    'month-array-northern': Array<number>;
    'month-array-southern': Array<number>;
  };
  shadow: string;
  price: number;
  'price-cj': number;
  icon_uri: string;
  image_uri: string;
}
