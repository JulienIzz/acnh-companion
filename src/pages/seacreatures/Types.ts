export type SeaCreature = {
  id: number;
  name: {
    'name-USen': string;
    'name-EUfr': string;
  };
  availability: {
    'month-northern': string;
    'month-southern': string;
    time: string;
    isAllDay: boolean;
    isAllYear: false;
    'month-array-northern': Array<number>;
    'month-array-southern': Array<number>;
    'time-array': Array<number>;
  };
  speed: string;
  shadow: string;
  price: number;
  image_uri: 'https://acnhapi.com/v1/images/sea/1';
  icon_uri: 'https://acnhapi.com/v1/icons/sea/1';
};
