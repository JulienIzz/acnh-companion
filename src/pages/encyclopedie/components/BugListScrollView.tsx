import {View, ScrollView, Image} from 'react-native';
import React from 'react';
import type {Bug} from '../Types';
import {SimpleTypography} from '../../../theme/typography/Typography';
import {styles, TabStackParamList} from '../Encyclopedie';
import {RouteProp} from '@react-navigation/native';

type BugListRouteProp = RouteProp<TabStackParamList, 'Bug'>;

type BugProps = {
  route: BugListRouteProp;
};

export function BugListScrollView({route}: BugProps) {
  const bugList: Bug[] = route.params.bugList;
  return (
    <ScrollView>
      {bugList.map(bug => (
        <View style={styles.animalCard}>
          <View style={styles.animalID}>
            <SimpleTypography text={'Name : ' + bug.name['name-USen']} />
            <SimpleTypography text={'Rarity : ' + bug.availability.rarity} />
            <SimpleTypography text={'Price : ' + bug.price + ' bells'} />
          </View>
          <View style={styles.animalImageContainer}>
            <Image
              style={styles.animalImage}
              source={{
                uri: bug.icon_uri,
              }}
            />
          </View>
        </View>
      ))}
    </ScrollView>
  );
}
