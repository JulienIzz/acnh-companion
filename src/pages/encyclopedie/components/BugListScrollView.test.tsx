import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {Animal} from '../Types';
import {BugListScrollView} from './BugListScrollView';
import {RouteProp} from '@react-navigation/native';
import {mockBugArray} from '../../../tests/mockBugArray';

type mockTabStackParamList = {
  Bug: {bugList: Animal[]};
};

const mockRoute: RouteProp<mockTabStackParamList, 'Bug'> = {
  key: 'Bug',
  name: 'Bug',
  params: {
    bugList: mockBugArray,
  },
};

test('fish scroll view card with 2 mock bug information', async () => {
  render(<BugListScrollView route={mockRoute} />);

  const firstFishNameOutput = screen.getByText('Nom : InsectePilote1');
  const firstFishPriceOutput = screen.getByText('Prix : 900');
  const firstFishRarityOutput = screen.getByText('Rareté : Common');
  const firstFishMonthsOutput = screen.getByText(
    'Disponibilité : Avril à Septembre',
  );
  const firstFishHoursOutput = screen.getByText('Horaire  : 9am - 4pm');
  const firstFishHocationOutput = screen.getByText('Localisation : River');
  const secondFishNameOutput = screen.getByText('Nom : InsectePilote2');
  const secondFishPriceOutput = screen.getByText('Prix : 900');
  const secondFishRarityOutput = screen.getByText('Rareté : Common');
  const secondFishMonthsOutput = screen.getByText(
    'Disponibilité : Avril à Septembre',
  );
  const secondFishHoursOutput = screen.getByText('Horaire  : 9am - 4pm');
  const secondFishLocationOutput = screen.getByText('Localisation : River');

  expect(firstFishNameOutput).toBeTruthy();
  expect(firstFishPriceOutput).toBeTruthy();
  expect(firstFishRarityOutput).toBeTruthy();
  expect(firstFishMonthsOutput).toBeTruthy();
  expect(firstFishHoursOutput).toBeTruthy();
  expect(firstFishHocationOutput).toBeTruthy();
  expect(secondFishNameOutput).toBeTruthy();
  expect(secondFishPriceOutput).toBeTruthy();
  expect(secondFishRarityOutput).toBeTruthy();
  expect(secondFishMonthsOutput).toBeTruthy();
  expect(secondFishHoursOutput).toBeTruthy();
  expect(secondFishLocationOutput).toBeTruthy();

  expect(screen.toJSON()).toMatchSnapshot();
});
