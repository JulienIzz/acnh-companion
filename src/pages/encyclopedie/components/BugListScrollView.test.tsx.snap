// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fish scroll view card with 2 mock bug information 1`] = `
<RCTScrollView
  data={
    [
      {
        "availability": {
          "isAllDay": false,
          "isAllYear": false,
          "location": "River",
          "month-array-northern": [
            4,
            5,
            6,
            7,
            8,
            9,
          ],
          "month-array-southern": [
            4,
            5,
            6,
            7,
            8,
            9,
          ],
          "month-northern": "Avril à Septembre",
          "month-southern": "Avril à Septembre",
          "rarity": "Common",
          "time": "9am - 4pm",
          "time-array": [
            9,
            10,
            11,
            12,
            13,
            14,
            15,
            16,
          ],
        },
        "icon_uri": "",
        "id": 1,
        "image_uri": "",
        "name": {
          "name-EUfr": "InsectePilote1",
          "name-USen": "InsectePilote1",
        },
        "price": 900,
        "price-cj": 1200,
      },
      {
        "availability": {
          "isAllDay": true,
          "isAllYear": true,
          "location": "Sea",
          "month-array-northern": [
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
            10,
            11,
            12,
          ],
          "month-array-southern": [
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
            10,
            11,
            12,
          ],
          "month-northern": "Tout l'année",
          "month-southern": "Tout l'année",
          "rarity": "Rare",
          "time": "Tout la journée",
          "time-array": [
            0,
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
            10,
            11,
            12,
            13,
            14,
            15,
            16,
            17,
            18,
            19,
            20,
            21,
            22,
            23,
          ],
        },
        "icon_uri": "",
        "id": 2,
        "image_uri": "",
        "name": {
          "name-EUfr": "InsectePilote2",
          "name-USen": "InsectePilote2",
        },
        "price": 15000,
        "price-cj": 18000,
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "alignContent": "center",
            "alignSelf": "center",
            "justifyContent": "center",
            "marginVertical": 5,
            "width": "90%",
          }
        }
      >
        <View
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 10,
              "width": "100%",
              "zIndex": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FAEDCD",
                "borderColor": "#D4A373",
                "borderRadius": 10,
                "borderWidth": 3,
                "flexDirection": "row",
                "padding": 8,
                "width": "100%",
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                  "justifyContent": "space-evenly",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "black",
                    "fontSize": 14,
                  }
                }
              >
                Nom : InsectePilote1
              </Text>
              <Text
                style={
                  {
                    "color": "black",
                    "fontSize": 14,
                  }
                }
              >
                Rareté : Common
              </Text>
              <Text
                style={
                  {
                    "color": "black",
                    "fontSize": 14,
                  }
                }
              >
                Prix : 900
              </Text>
            </View>
            <View
              style={
                {
                  "height": 80,
                  "margin": 5,
                  "width": 80,
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "",
                  }
                }
                style={
                  {
                    "height": "100%",
                    "width": "100%",
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            {
              "alignSelf": "center",
              "backgroundColor": "#FAEDCD",
              "borderBottomColor": "#D4A373",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderBottomWidth": 3,
              "borderLeftColor": "#D4A373",
              "borderLeftWidth": 3,
              "borderRightColor": "#D4A373",
              "borderRightWidth": 3,
              "height": 100,
              "justifyContent": "space-evenly",
              "marginTop": 0,
              "padding": 5,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
                "fontSize": 14,
              }
            }
          >
            Disponibilité : Avril à Septembre
          </Text>
          <Text
            style={
              {
                "color": "black",
                "fontSize": 14,
              }
            }
          >
            Horaire : 9am - 4pm
          </Text>
          <Text
            style={
              {
                "color": "black",
                "fontSize": 14,
              }
            }
          >
            Localisation : River
          </Text>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "alignContent": "center",
            "alignSelf": "center",
            "justifyContent": "center",
            "marginVertical": 5,
            "width": "90%",
          }
        }
      >
        <View
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 10,
              "width": "100%",
              "zIndex": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FAEDCD",
                "borderColor": "#D4A373",
                "borderRadius": 10,
                "borderWidth": 3,
                "flexDirection": "row",
                "padding": 8,
                "width": "100%",
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                  "justifyContent": "space-evenly",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "black",
                    "fontSize": 14,
                  }
                }
              >
                Nom : InsectePilote2
              </Text>
              <Text
                style={
                  {
                    "color": "black",
                    "fontSize": 14,
                  }
                }
              >
                Rareté : Rare
              </Text>
              <Text
                style={
                  {
                    "color": "black",
                    "fontSize": 14,
                  }
                }
              >
                Prix : 15000
              </Text>
            </View>
            <View
              style={
                {
                  "height": 80,
                  "margin": 5,
                  "width": 80,
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "",
                  }
                }
                style={
                  {
                    "height": "100%",
                    "width": "100%",
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            {
              "alignSelf": "center",
              "backgroundColor": "#FAEDCD",
              "borderBottomColor": "#D4A373",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderBottomWidth": 3,
              "borderLeftColor": "#D4A373",
              "borderLeftWidth": 3,
              "borderRightColor": "#D4A373",
              "borderRightWidth": 3,
              "height": 100,
              "justifyContent": "space-evenly",
              "marginTop": 0,
              "padding": 5,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
                "fontSize": 14,
              }
            }
          >
            Disponibilité : Tout l'année
          </Text>
          <Text
            style={
              {
                "color": "black",
                "fontSize": 14,
              }
            }
          >
            Horaire : Tout la journée
          </Text>
          <Text
            style={
              {
                "color": "black",
                "fontSize": 14,
              }
            }
          >
            Localisation : Sea
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
